wagtail/models/__init__.py

```diff
@@ -535,6 +535,14 @@ def get_latest_revision_as_object(self):
         else:
             return self
 
+    @cached_property
+    def scheduled_revision(self):
+        return self.revisions.filter(approved_go_live_at__isnull=False).first()
```

Comment:
This and all model changes – makes sense to me, but I’d prefer someone else reviews considering those are quite fundamental APIs.